### PR TITLE
fix device status problem with device's namespace in kubeege v1.16.0

### PR DESCRIPTION
**Pull request by luomengY**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by luomengY** [comment]
This pr（https://github.com/kubeedge/kubeedge/pull/5396 ） cannot be reopened, so I created a new one. @wbc6080 @WillardHu

File: cloud/pkg/common/messagelayer/util.go [hunk 1]
```diff
@@ -131,8 +131,8 @@ func BuildResourceForDevice(nodeID, resourceType, resourceID string) (resource s
 // GetDeviceID returns the ID of the device
 func GetDeviceID(resource string) (string, error) {
 	res := strings.Split(resource, "/")
-	if len(res) >= ResourceDeviceIDIndex+1 && res[ResourceDeviceIndex] == ResourceDevice {
-		return res[ResourceDeviceIDIndex], nil
+	if len(res) >= ResourceDeviceIDIndex+2 && res[ResourceDeviceIndex] == ResourceDevice {
```
Comment (WillardHu) [inline cloud/pkg/common/messagelayer/util.go:134]: You can define some constants to replace magic numbers or add some comments.

Comment (luomengY) [reply cloud/pkg/common/messagelayer/util.go:134]: ok，I have made it.

**Comment by wbc6080** [comment]
@Shelley-BaoYue  @zhiyingfang2022  Please help add this pr to version 1.17, thanks

**Comment by zhiyingfang2022** [comment]
> @Shelley-BaoYue @zhiyingfang2022 Please help add this pr to version 1.17, thanks

ok,done.

**Review by wbc6080 — COMMENTED** [review]
/lgtm
It need to cherry-pick to release-1.16 after merged

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**